### ISSUE TITLE
How long is "a long time"? / lack of log messages output to console

### ISSUE THREAD

**Issue by splendifer** [mit-nlp/MITIE#177]
Hi,

Based on the log posted by autopost-get, it seems like it's normal to get log messages like

```
C: 20 loss: 3 0.431569
C: 35 loss: 3 0.437956
C: 20 loss: 4.5 0.443431
C: 5 loss: 3 0.440693
```

while the training is going on. However, for over ~five hours, there hasn't been any message output; the complete entirety of the log consists of 

```
words in dictionary: 200000
num features: 271
now do training
C:           20
epsilon:     0.01
num threads: 4
cache size:  5
loss per missed segment:  3
```

and has not updated for those five hours. Is this to be expected? My training dataset consists of fewer than 30 thousand sentences -- is that far too much? If so, what is a more appropriate scale to work at?

Thank you for your time!

**Comment by davisking** [comment]
Be patient.  MITIE does a lot of rounds of training and the whole search is
very conservative.  That is, it errs on the side of running much longer to
ensure it finds a good solution.  The training time is linear in the
dataset size and it can learn from very large datasets, much larger than
30k sentences.  However, there can be very large fixed overhead due to the
underlying difficulty of your training data.  This has nothing to do with
the size of the dataset.  For instance, if your samples are mislabeled or
the labels are ambiguous then the hyperparameter search will often take
much longer, regardless of the training set size.

**Comment by splendifer** [comment]
I see, thank you! Is the place where my run has stalled (right after the first "loss per missed segment" message) where the hyperparameter search is taking place?

**Comment by davisking** [comment]
Yes

**Comment by splendifer** [comment]
Okay, good to know what's going on. I will double-check the data. Thanks!
